README.md

```diff
@@ -3,58 +3,136 @@
 
-> This section is still a work-in-progress and will be changed as we continue to move in files from other repos.
 
-Daily test passes are scheduled by Prow ([see: config](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/sig-windows/sig-windows-config.yaml)), and results are on [TestGrid](https://testgrid.k8s.io/sig-windows).
+If you're new to building and testing Kubernetes, it's probably best to read the official [End-to-End Testing in Kubernetes](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/e2e-tests.md) page first. The rest of this page has a summary of those steps tailored to testing clusters with Windows nodes.
 
-### Cluster Setup
 
+## Building Tests
+
+### e2e.test
+
+The official steps are in [kubernetes/community](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/e2e-tests.md#building-kubernetes-and-running-the-tests). For more details, be sure to read that doc. This is just a short summary.
+
+Make sure you have a working [Kubernetes development environment](https://github.com/kubernetes/community/blob/master/contributors/devel/development.md) on a Mac or Linux machine. If you're using Windows, you can use WSL, but it will be slower than a Linux VM. The tests can be run from the same VM, as long as you have a working KUBECONFIG.
+
+```bash
+go get -d k8s.io/kubernetes
+cd $GOPATH/src/k8s.io/kubernetes
+./build/run.sh make WHAT=test/e2e/e2e.test
+```
+Once complete, the binary will be available at: `~/go/src/k8s.io/kubernetes/_output/dockerized/bin/linux/amd64/e2e.test`
+
+#### Cross-building for Mac or Windows
+
+To build a binary to run on Mac or Windows, you can add `KUBE_BUILD_PLATFORMS`.
+
+For Windows
+```bash
+./build/run.sh make KUBE_BUILD_PLATFORMS=windows/amd64 WHAT=test/e2e/e2e.test
+```
+
+For Mac
+```bash
+./build/run.sh make KUBE_BUILD_PLATFORMS=darwin/amd64 WHAT=test/e2e/e2e.test
+```
+
+Your binaries will be available at `~/go/src/k8s.io/kubernetes/_output/dockerized/bin/linux/amd64/e2e.test` where `linux/amd64/` is replaced by `KUBE_BUILD_PLATFORMS` if you are building on Mac or Windows.
+
+
+## Running an e2e test pass
+
+
+### Using an existing cluster
 
-### Running SIG-Windows tests
+All of the tests are built into the `e2e.test` binary, which you can as a standalone binary to test an existing cluster.
 
-All of the SIG-Windows tests are included in the `e2e.test` binary. You need to set a few options to connect to the cluster, and use the right Windows images.
+There are a few important parameters that you need to use:
+
+- `--provider=local` - this will avoid using a cloud provider to provision new resources such as storage volumes or load balancers
```

Comment:
We should mention that with provider=local test should be run from the master node as settings here: https://github.com/kubernetes/kubernetes/blob/master/cluster/local/util.sh#L22 will default the master ip to localhost. 

To run tests from any machine, we should use --provider=skeleton and export: KUBE_MASTER_IP= and KUBE_MASTER_URL="http://${KUBE_MASTER_IP}:8080" as well.

Reply:
`skeleton` might work just fine for all use cases?
skeleton is the default bare-metal (null) provider that does not depend on the /cluster folder.

i don't know all the differences between skeleton and local but two i know about are logging and ssh handling that comes from bash scripts from the cluster folder. only local has that.

i see the ACS deployer is not using https://github.com/kubernetes/kubernetes/blob/master/cluster/log-dump/log-dump.sh which is the default way to dump cluster logs
https://github.com/kubernetes/test-infra/blob/master/kubetest/azure.go#L580-L581

please mind that the cluster folder is going away so "local" is going to be broken eventually, so migrating to "skeleton" now is better.

Reply:
Yes, skeleton is fine for all cases.

Reply:
Thanks! Updating now. I'll check on the log-dump for a future PR

Reply:
if you already have a different method for log dumping https://github.com/kubernetes/test-infra/blob/master/kubetest/azure.go#L580-L581 better remain no-op to not use defaultDumpClusterLogs() (which uses log-dump.sh behind the scenes).

as log-dumps.sh is deprecated.

Reply:
If it's deprecated, what replaces it?

Reply:
the kops deployer has an example of what is the replacement:
https://github.com/kubernetes/test-infra/blob/master/kubetest/kops.go#L452-L499
(newLogDumper)

but i couldn't get it to work well in the anywhere.go deployer.

Reply:
@PatrickLang  @neolit123  Yeah, that's what I've tested with, writing to localPath variable folder, using logslurp + some custom scripts. However that is a little bit complicated. With ssh enabledby default in Windows, it should be easier.